File: scripts/parse-lint.py
import re
import sys

with open(sys.argv[1], encoding='utf-8') as f:
    lines = f.readlines()

current = ''
counts = {}
for line in lines:
    if 'mobile:lint: D:' in line and 'src' in line:
        m = re.search(r'(apps[/\\\\]mobile[/\\\\]src[/\\\\][^\s]+)', line)
        if m:
            current = m.group(1).replace('\\', '/')
    if 'no-unused-vars' in line:
        counts[current] = counts.get(current, 0) + 1

for f, c in sorted(counts.items(), key=lambda x: -x[1]):
    print(f"{c}\t{f}")

File: scripts/parse-lint.mjs
import { readFileSync } from 'fs';

const input = readFileSync(0, 'utf8');
const lines = input.split('\n');

let currentFile = '';
const counts = {};

for (const line of lines) {
  if (line.includes('mobile:lint: D:') && line.includes('src')) {
    const idx = line.indexOf('apps');
    if (idx !== -1) {
      currentFile = line.substring(idx).split(/\s{2,}/)[0].trim();
    }
  }
  if (line.includes('no-unused-vars') && currentFile) {
    counts[currentFile] = (counts[currentFile] || 0) + 1;
  }
}

const sorted = Object.entries(counts).sort((a, b) => b[1] - a[1]);
console.log('Count\tFile');
for (const [file, count] of sorted) {
  console.log(`${count}\t${file}`);
}
console.log(`\nTotal: ${sorted.reduce((s, [, c]) => s + c, 0)}`);
